File: framework/wcf/Scenario/Ajax/WeaklyTypedJson/VB/WeaklyTypedJson/ClientSideService.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved. 
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System.ServiceModel
Imports System.ServiceModel.Channels
Imports System.ServiceModel.Web

Namespace Microsoft.Samples.WeaklyTypedJson

    <ServiceContract()> _
    Friend Interface IClientSideProfileService
        ' There is no need to write a DataContract for the complex type returned by the service.
        ' The client will use a JsonObject to browse the JSON in the received message.

        <WebGet(ResponseFormat:=WebMessageFormat.Json)> _
        Function GetMemberProfile() As Message
    End Interface

End Namespace

File: framework/wcf/Scenario/Ajax/WeaklyTypedJson/CS/WeaklyTypedJson/ClientSideService.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System.ServiceModel;
using System.ServiceModel.Channels;
using System.ServiceModel.Web;

namespace Microsoft.Samples.WeaklyTypedJson
{
    [ServiceContract]
    interface IClientSideProfileService
    {
        // There is no need to write a DataContract for the complex type returned by the service.
        // The client will use a JsonObject to browse the JSON in the received message.

        [WebGet(ResponseFormat = WebMessageFormat.Json)]
        Message GetMemberProfile();
    }
}
